pyccel/stdlib/ndarrays/ndarrays.h

```diff
@@ -95,20 +101,24 @@ typedef struct  s_ndarray
     enum e_types            type;
     /* type size of the array elements */
     int32_t                 type_size;
-    /* number of element in the array */
+    /* max number of element in the array */
     int32_t                 length;
+    /* number of element currently in the array (for the copy operations)*/
+    int32_t                 current_length;
```

Comment:
Why does this need to be a member of the class that exists after the copy? It looks like it ought to be a temporary given it's definition

Reply:
Makes sense, it would mean we would have to keep track of what's being copied and update `offset` in the middle of the code, for something like `[[a, b], [[1, 2, 3], [4, 5, 6]], [[7, 8, 9], c]]`, where the middle portion (`[[1, 2, 3], [4, 5, 6]], [[7, 8, 9]`) should be copied in a single go, that wouldn't add clutter to the code?

Reply:
`offset` was used before this, but it used to be easily calculated using `sizeof(array) * position`, the problem with this is that for somthing like `[a, b, [1, 2, 3], [7, 8, 9], [5, 6, 7]]`, it would make 5 copy operations instead of just 3, since each of the scalar arrays would be copied independently

Reply:
I wouldn't get rid of `current_length` I would just create it locally while doing the copies if and when necessary rather than keeping for as long as the array exists

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/2a388b28d73190aa7333b30a0e0a4d7cb963bc29

Reply:
@EmilyBourne